static/app/components/compactSelect/control.tsx

```diff
@@ -313,6 +320,18 @@ export function Control({
     },
   });
 
+  /**
+   * The menu's full width, before any option has been filtered out. Used to maintain a
+   * constant width while the user types into the search box.
+   */
+  const [menuFullWidth, setMenuFullWidth] = useState<number>();
```

Comment:
I wonder if setting the width on open could lead to other unintended behavior? What if the menu is small on first render and then more options get populated that might want a larger overlay?

Is it possible to just set a min-width on the overlay instead?

Reply:
Hmm that's a good point. I updated the code to only "lock" the menu's width while the search input is focused (d30ba854856865cd0fca204a3c3834f8d22aac1b). This should help prevent unintended behaviors: if the options list gets updated while the search input is blurred (which is most of the time), then the menu is still free to expand/contract. While the search box is focused, the menu's width is locked to prevent a jarring search experience.